lib/ledger_sync/adaptors/netsuite/department/searcher.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module LedgerSync
+  module Adaptors
+    module NetSuite
+      module Department
+        class Searcher < Searcher
+          def resources
+            @resources ||= begin
+              @request = adaptor
+                .get(
+                  path: "/department?limit=#{limit}&offset=#{offset}"
+                )
+
+              request.body
```

Comment:
We can probably DRY this up with `inferred_resource_class` later on, seeing as it looks like most searchers follow this pattern.  Maybe even above, though I imagine we will need to make searchers more advanced (e.g. handle different attributes) at some point.

Reply:
Added to #168